src/cache/ElasticacheRedis.spec.ts

```diff
@@ -1,142 +1,160 @@
-import { ElasticacheRedis, RedisClient } from './ElasticacheRedis';
-import { KeyValueCacheSetOptions } from './interface';
+import { afterEach, expect, jest, describe, it } from '@jest/globals';
+import { ElasticacheRedis } from './ElasticacheRedis';
+// import { KeyValueCacheSetOptions } from '@apollo/utils.keyvaluecache';
 
-class FakePrimaryClient implements RedisClient {
-  client: any;
+import Redis from 'ioredis-mock';
 
-  clear(): Promise<void> {
-    return Promise.resolve(undefined);
-  }
+/**
+ * Use of ioredis-mock might seem overkill for validation, but it works.
+ *
+ * Do not try to mock ioredis at a library level with something like
+ * jest.Mock<Redis>. even commands like `get` and `set` are generated
+ * at runtime.
+ */
 
-  close(): Promise<void> {
-    return Promise.resolve(undefined);
-  }
-
-  delete(key: string): Promise<boolean> {
-    return Promise.resolve(false);
-  }
-
-  flush(): Promise<void> {
-    return Promise.resolve(undefined);
-  }
-
-  get(key: string): Promise<string | undefined> {
-    return Promise.resolve(undefined);
-  }
-
-  set(
-    key: string,
-    value: string,
-    options: KeyValueCacheSetOptions | undefined
-  ): Promise<void> {
-    return Promise.resolve();
-  }
-}
-
-class FakeReaderClient extends FakePrimaryClient {}
-
-const redis = new ElasticacheRedis(
-  new FakePrimaryClient(),
-  new FakeReaderClient()
-);
+const cache = new ElasticacheRedis(new Redis(), new Redis());
 
 describe('ElasticacheRedis', () => {
-  afterEach(() => jest.clearAllMocks());
+  afterEach(async () => jest.clearAllMocks());
 
   it('can get cache key', async () => {
-    expect(redis.getKey('test')).toEqual('098f6bcd4621d373cade4e832627b4f6');
+    expect(cache.getKey('test')).toEqual('098f6bcd4621d373cade4e832627b4f6');
   });
 
   it('can get cache value', async () => {
-    const readerGet = (FakeReaderClient.prototype.get = jest.fn());
-
-    await redis.get('test');
-
-    expect(readerGet).toHaveBeenCalledTimes(1);
-    expect(readerGet).toHaveBeenCalledWith('test');
+    const readerData = { test: '1' };
+    const cache = new ElasticacheRedis(
+      new Redis(),
+      new Redis({ data: readerData })
+    );
+
+    const readerGetSpy = jest.spyOn((cache as any).readerClient, 'get');
+    const primaryGetSpy = jest.spyOn((cache as any).primaryClient, 'get');
```

Comment:
these are repeated, so if you wanted to simplify your code a little bit you could pull it out of the scope and use the before/after hooks to flush the history 

but that's more a stylistic thing